app/models/miq_server/worker_management/monitor/kubernetes.rb

```diff
@@ -15,33 +20,88 @@ def failed_deployments(restart_count = 5)
     current_pods.values.select { |h| h[:last_state_terminated] && h.fetch(:container_restarts, 0) > restart_count }.collect { |h| h[:label_name] }
   end
 
+  def sync_deployment_settings
+    return unless podified?
+    return unless Settings.server.worker_monitor.enforce_resource_constraints
+
+    checked_deployments = Set.new
+    podified_miq_workers.each do |worker|
+      next if checked_deployments.include?(worker.worker_deployment_name)
+
+      if deployment_resource_constraints_changed?(worker)
+        _log.info("Constraints changed, patching deployment: [#{worker.worker_deployment_name}]")
+        worker.patch_deployment
+      end
+      checked_deployments << worker.worker_deployment_name
+    end
+  end
+
+  def podified_miq_workers
+    # Cockpit is a threaded worker in the orchestrator that spins off a process it monitors and isn't a pod worker.
+    miq_workers.where.not(:type => %w[MiqCockpitWsWorker])
+  end
+
+  def deployment_resource_constraints_changed?(worker)
+    container = current_deployments.fetch_path(worker.worker_deployment_name, :spec, :template, :spec, :containers).try(:first)
+    current_constraints = container.try(:fetch, :resources, nil) || {}
+    desired_constraints = worker.resource_constraints
+    constraints_changed?(current_constraints, desired_constraints)
+  end
+
+  def constraints_changed?(current, desired)
+    return true if (current.blank? && desired.present?)
+    return false if (current.blank? && desired.blank?)
+
+    !cpu_value_eql?(current.fetch_path(:requests, :cpu), desired.fetch_path(:requests, :cpu)) ||
+      !cpu_value_eql?(current.fetch_path(:limits, :cpu), desired.fetch_path(:limits, :cpu)) ||
+      current.fetch_path(:requests, :memory).try(:iec_60027_2_to_i) != desired.fetch_path(:requests, :memory).try(:iec_60027_2_to_i) ||
+      current.fetch_path(:limits, :memory).try(:iec_60027_2_to_i)   != desired.fetch_path(:limits, :memory).try(:iec_60027_2_to_i)
+  end
+
   private
 
-  def start_pod_monitor
+  def cpu_value_eql?(current, desired)
+    # Convert "1" to "1000m" (millicores) so "1" == "1000m"
```

Comment:
The comment is backwards

```suggestion
    # Convert "1000m" (millicores) to "1" for comparisons.
```

Reply:
Oh no wait, it's not backwards..you are converting to millicores but as an integer... so not litearlly to `"1000m"`, but to `1000`

Reply:
This should be resolved